extensions/markdown-language-features/package.json

```diff
@@ -26,6 +26,15 @@
 		"onView:markdown.preview"
 	],
 	"contributes": {
+		"webviews": [
```

Comment:
I initially tried putting this new point under `views`, but now that we support custom view containers it does not seem like a great fit there

Reply:
Not sure what you mean by 'custom view containers'?

Reply:
The custom activity bar sections that we added last iteration. I see two concerns with reusing `views` for webview contributions:

* Webview contributions would not have the same properties as the other items in `views`.
* As we now support contributing a custom view container with any name, there's nothing to stop someone from creating a view container named `webviews`.  Pretty unlikely this will be a real problem so the first point is the main concern

Reply:
If it has to be merged into the `views` extension point, then `editor` is the right location for web views instead of `webviews`. IMO Web view and Tree view are types of views. I see that at present Tree views and Web views do not have common schema, for e.g. Tree views do not have icons where as web views do. We can have either schema by type or schema by location depends on how they emerge in future. Looking at present scenario, it looks to me that the `location` drives the properties. I mean, views registered in activity bar containers do not need icons but icons are needed for views registered under editor. In future, if we want to be flexible in moving views across, then all views can define icons optionally. Not sure what is `viewType` property in web views. May be this can be used to differentiate tree views vs web views?


I think the second point you mentioned is a valid concern. May be we should think of restricting some default locations like `editor` or `panel`.

Or other approach would be to use viewsContainers` extension point? Say, introduce a new location `editor` and create a new view container for web views. It means a view container inside editor can contain multiple views(web or tree).